examples/image_filters_using_image2d_t.py

```diff
@@ -0,0 +1,242 @@
+                                       write_only image2d_t dest,
+                                       const sampler_t sampler,
+                                       const float scalar,
+                                       const float4 mean_intensity){
+  int2 pos = (int2)(get_global_id(0),get_global_id(1));
+  uint4 pixel = read_imageui(src,sampler,pos);
+  uint4 new_pixel = convert_uint4((float4)scalar*(convert_float4(pixel)\
+                                        -mean_intensity)+mean_intensity);
+  uint4 overflow = convert_uint4(new_pixel>(uint4)255);
+  write_imageui(dest,pos,select(new_pixel,(uint4)255,overflow));
+}
+__kernel void reduce_sum(__global int *in,
+            __global int *reduce,
+            __local int *buffer)
+{
+  uint gid = get_global_id(0);
+  uint wid = get_group_id(0);
+  uint lid = get_local_id(0);
+  uint gs = get_local_size(0);
+  buffer[lid] = in[gid];
+
+  for(uint s = gs/2; s > 0; s >>= 1) {
+    barrier(CLK_LOCAL_MEM_FENCE);
+    if(lid < s) {
+      buffer[lid] += buffer[lid+s];
+    }
+  }
+  if(lid == 0) reduce[wid] = buffer[lid];
+}
+__kernel void gaussian_blur(read_only const image2d_t src,
+                                       write_only image2d_t dest,
+                                       const sampler_t sampler,
+                                       read_only const image2d_t gauss_kernel,
+                                       const short dim){
+  int2 pos = (int2)(get_global_id(0),get_global_id(1));
+  uint4 pixel=(uint4)0;
+  float4 src_pixel,gauss_pixel;
+  for(int i = -dim/2;i<dim/2;i++){
+    for(int j = -dim/2;j<dim/2;j++){
+      src_pixel=convert_float4(read_imageui(src,sampler,(int2)(i,j)+pos));
+      gauss_pixel=convert_float4(read_imageui(gauss_kernel,sampler,(int2)(i+dim/2,j+dim/2)))/(float4)255;
+      pixel=pixel+convert_uint4(src_pixel*gauss_pixel);
+      if(get_global_id(0)==0&&get_global_id(1)==0){
+        printf("i=%d,j=%d",i,j);
+      }
+    }
+  }
+  uint4 overflow = convert_uint4(pixel>(uint4)255);
+  write_imageui(dest,pos,select(pixel,(uint4)255,overflow));
+}
+""",
+).build()
+# image_path = input("Enter path to the image:")
+image_path = "./noisyImage.jpg"
+try:
+    # intel-compile-runtime supports RGBA so im using that format
+    image = np.array(Image.open(image_path).convert("RGBA"))
+except FileNotFoundError:
+    print("Image does not exist")
+    sys.exit()
+except EOFError:
+    print("No input provided")
+    sys.exit()
+
+src_buf = cl.image_from_array(context, image, 4)
+
+image_format = cl.ImageFormat(
+    cl.channel_order.RGBA, cl.channel_type.UNSIGNED_INT8)
+
+shape = (image.shape[1], image.shape[0])
+
+dest_buf = cl.create_image(
+    context, cl.mem_flags.WRITE_ONLY, image_format, shape)
+sampler = cl.Sampler(
+    context, False, cl.addressing_mode.CLAMP_TO_EDGE, cl.filter_mode.NEAREST
+)
+
+print("Enter one of the next values\n")
+# i = int(input("Gray scale:1\nGaussian blur:2\nBrightenss adjustment:3\n"))
+i = 2
+if i in range(1, 4):
```

Comment:
I would also suggest removing this `if` check for the values of `i`. Since we hardcode this in the example, it should be fine to simply it.